packages/chat-component-bindings/src/utils/getParticipantsWhoHaveReadMessage.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { ChatMessageWithStatus } from '@internal/chat-stateful-client';
+
+/**
+ * @private
+ *logic: Looking at message A, how do we know it's read number?
```

Comment:
I'm a bit confused by this logic.

`message.createdOn` is set when a message arrives at the ACS backend, this is before the message has been sent to a client.
`readReceipt.readOn` is set when we explicitly send a read receipt from a client.

So, it is possible that a message was sent by user 1, and user 2 sent a read receipt after that, but before it received the message from user 1. In that case, we will incorrectly report that user 2 has read the message.

Should we instead send read receipt for each message individually? Nothing in the headless SDK API seems to block us from doing that, right?

Reply:
In this case since our assumption is : if user has read a message, user has read all messages before that. we can safely say that if a message is created before the readon time, the message is read

Reply:
readreceipt won't be sent until a message is received and seen. So the problem should not occur

Reply:
First, I was wrong -- both timestamps are computed at the backend server. That's good, so we can assume sync clocks.

Reply:
But the original concern stands:

    User1                              Server                                     User2
    ----------------------------
    Sends Message 1
                                          Message1:timestamp1
    Sends Message 2      
                                          Message2:timestamp2
                                                                                          Gets message1, swends readReceipt1
                                          ReadReceipt1:timestamp3
    Gets readReceipt1:timestamp3
                                                                                          Gets message2

Here, readReceipt1 has timestamp3 > message2's timestamp2. So User1 is shown that User2 has read message2. But that is wrong.

Reply:
Let's bring this discussion to the team channel. We can create a new work item to track fixing this race condition properly.

This PR also adds the UI surfaces for showing the detailed read report, so I don't want to block this PR on the discussion.

Reply:
Ok sounds good! Thanks!